### PR TITLE
Skip artifacts from adding them to the scanner that do not exist.

### PR DESCRIPTION
**Pull request by twwwt**
Proposed solution to issue #1631

### CODE REVIEW

**Comment by twwwt** [comment]
Rethinking that pull request, I think it should **not** be applied. The reason is that the behavior of skipping an artifact that does not exist should take place as late as possible; that is, the moment when the scanner actually runs. The pull request implements eager behavior: an artifact is not added if it does not exist at Gradle's configuration phase. In general, an artifact may very well be created only after configuration phase.

In summary, the additional test whether a file/artifact does actually exist should be implemented directly inside Flyway's scanner, not in the Gradle plugin.

**Comment by axelfontaine** [comment]
@twwwt I hadn't looked at the code yet. Agreed.
